charts/hivemq-edge/templates/statefulset.yml

```
@@ -0,0 +1,187 @@
+              valueFrom:
+                secretKeyRef:
+                  name: "{{ .Values.mqtts.keystore.passwordSecretName }}"
+                  key: "{{ .Values.mqtts.keystore.passwordSecretKey }}"
+            - name: HIVEMQ_MQTTS_SECRET_PRIVATE_KEY_PASSWORD
+              valueFrom:
+                secretKeyRef:
+                  name: "{{ .Values.mqtts.keystore.privateKeyPasswordSecretName }}"
+                  key: "{{ .Values.mqtts.keystore.privateKeyPasswordSecretKey }}"
+            {{- end }}
+            {{- if ne .Values.mqttsClientauth.clientAuthenticationMode "NONE" }}
+            - name: HIVEMQ_MQTTS_TRUSTSTORE_PATH
+              value: /mqtts/truststore.jks
+            - name: HIVEMQ_MQTTS_SECRET_TRUSTSTORE_PASSWORD
+              valueFrom:
+                secretKeyRef:
+                  name: "{{ .Values.mqttsClientauth.truststore.passwordSecretName }}"
+                  key: "{{ .Values.mqttsClientauth.truststore.passwordSecretKey }}"
+            {{- end }}
+            {{- with .Values.env }}
+              {{- toYaml . | nindent 12 }}
+            {{- end }}
+          resources:
+            limits:
+              memory: "{{ .Values.resources.memory }}"
+              cpu: "{{ .Values.resources.cpu }}"
+            requests:
+              memory: "{{ .Values.resources.memory }}"
+              cpu: "{{ .Values.resources.cpu }}"
+          volumeMounts:
+          - name: fragment
+            mountPath: /fragment
+          {{- range .Values.volumes }}
+          {{- with (index .definition 0) }}
+          - name: {{ .name }}
+          {{- end }}
+            mountPath: {{ .mount }}
+          {{- end }}
+          {{- if .Values.mqtts.enable }}
+          - mountPath: /mqtts/mqtts.jks
+            name: mqtts-jks
+            subPath: "{{ .Values.mqtts.keystore.keystoreSecretKey }}"
+          {{- end }}
+          {{- if .Values.https.enable }}
+          - mountPath: /https/keystore.jks
+            name: https-jks
+            subPath: "{{ .Values.https.keystore.keystoreSecretKey }}"
+          {{- end }}
+          {{- if ne .Values.mqttsClientauth.clientAuthenticationMode "NONE" }}
+          - mountPath: /mqtts/truststore.jks
+            name: mqtts-truststore-jks
+            subPath: "{{ .Values.mqttsClientauth.truststore.truststoreSecretKey }}"
+          {{- end }}
+          {{- if .Values.license.enabled }}
+          - mountPath: /license.edgelic
+            name: license
+          {{- end }}
+          {{- if .Values.persistence.enabled }}
+          - mountPath: /persistence
+            name: persistence
+          {{- end }}
+      {{- with .Values.podScheduling.affinity }}
+      affinity:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      {{- with .Values.podScheduling.tolerations }}
+      tolerations:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      volumes:
+        - name: fragment
+          configMap:
+            name: hivemq-edge-config-edge
```

The ConfigMap name is hardcoded to `hivemq-edge-config-edge`. However the actual ConfigMap name is based on the `releaseName`, so this won't work for a Helm chart release with name different than `edge`:

```yaml
apiVersion: v1
kind: ConfigMap
metadata:
  labels:
    {{- include "hivemq-edge.labels" . | nindent 4 }}
  name: {{ include "hivemq-edge.name" (dict "prefix" "hivemq-edge" "name" "config" "releaseName" .Release.Name) }}
```

Something like this should work:
```suggestion
          configMap:
            name: {{ include "hivemq-edge.name" (dict "prefix" "hivemq-edge" "name" "config" "releaseName" .Release.Name) }}
```